kernel/thread.c

```diff
@@ -36,6 +36,38 @@ extern struct _static_thread_data _static_thread_data_list_end[];
 	     thread_data < _static_thread_data_list_end; \
 	     thread_data++)
 
+#if defined(CONFIG_THREAD_MONITOR)
+void k_thread_foreach(thread_user_cb user_cb, void *user_data)
```

Comment:
The argument user_cb should be a const.

Reply:
this is not a struct data. It is function pointer and the intention is not to be const function()